File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def CalculateTotalHours(self, piles, mid):
        totalHours = 0
        for val in piles:
            totalHours += math.ceil(val/mid)
        
        return totalHours


    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)

        while left<right:
            mid = left+(right-left) // 2
            totalHours = self.CalculateTotalHours(piles, mid)

            if totalHours <= h:
                right = mid
            else:
                left = mid+1
        return left

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int length  = prices.length;

        int profit = 0;
        for(int i = 0;i < length;i++){
            for(int j = i+1 ; j< length; j++){
                profit = Math.max(prices[j] - prices[i], profit);
            }
        }
        return profit;
    }
}
